### PR TITLE
Enhance FindInFiles - part 1

### PR DESCRIPTION
**Pull request by pnedev**
This pull request is re-based part 1 of this one: https://github.com/donho/notepad-plus-plus/pull/83
The patches in this part should be easily review-able and understandable.
Once this PR is merged I will make a new one that will comprise part 2 of the original PR.

### CODE REVIEW

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 1]
```diff
@@ -3005,21 +3016,22 @@ void Progress::setPercent(unsigned percent, const TCHAR *fileName) const
...
 		::SendMessage(_hPText, WM_SETTEXT, 0, (LPARAM)fileName);
 	}
 }
 
 
-void Progress::close()
+void Progress::flushCallerUserInput() const
 {
-	if (_hwnd)
+	MSG msg;
```
Comment (ariccio) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: What's `msg` doing out here? Does it need to persist across loop iterations? (I doubt it?)

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 2]
```diff
@@ -3105,10 +3113,25 @@ int Progress::createProgressWindow()
 
 RECT Progress::adjustSizeAndPos(int width, int height)
 {
-	RECT win, maxWin;
-
+	RECT maxWin;
```
Comment (ariccio) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: While you're at it, maybe it's a good idea to zero init `maxWin`, so we don't read random garbage if `GetWindowRect` fails?

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 3]
```diff
@@ -3129,9 +3154,11 @@ RECT Progress::adjustSizeAndPos(int width, int height)
 		win.right = maxWin.right;
 	}
 
-	if (height < maxWin.right - maxWin.left)
+	if (height < maxWin.bottom - maxWin.top)
```
Comment (ariccio) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: Nice catch.

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp (hunk 2 above)
Comment (MAPJe71) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: RECT maxWin = { 0 };

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 4]
```diff
@@ -3052,14 +3064,11 @@ int Progress::thread()
...
-	if (_hOwner)
-		styleEx |= WS_EX_TOOLWINDOW;
-
-	_hwnd = ::CreateWindowEx(styleEx,
+	_hwnd = ::CreateWindowEx(
+		WS_EX_TOOLWINDOW | WS_EX_OVERLAPPEDWINDOW | WS_EX_TOPMOST,
 		cClassName, _header, WS_POPUP | WS_CAPTION,
 		CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT,
-		_hOwner, NULL, _hInst, (LPVOID)this);
+		NULL, NULL, _hInst, (LPVOID)this);
```
Comment (ariccio) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: Can you drop the `(LPVOID)` cast? It [shouldn't be necessary](http://eli.thegreenplace.net/2009/11/16/void-and-casts-in-c-and-c).

Comment (ariccio) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: Xref: #125

**Pull request merged**